IonDotnet/Internals/Lite/IonValueLite.cs

```diff
@@ -169,6 +169,22 @@ protected bool IsSystemValue(bool value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         protected bool IsNullValue() => HasFlag(NullFlag);
 
+        protected void ValidateThisNotNull()
+        {
+            if (IsNull)
+            {
+                throw new ArgumentException("The value is null");
```

Comment:
Please see and use this [exception ](https://github.com/dhhoang/IonDotnet/blob/master/IonDotnet/NullValueException.cs)